HLTriggerOffline/Scouting/plugins/ScoutingEGammaCollectionMonitoring.cc

```diff
@@ -0,0 +1,459 @@
+/*
+Scouting EGamma DQM core implementation.
+
+ Description: ScoutingEGammaCollectionMonitoring is developed to enable us to monitor the
+              comparison between pat::Object and Run3Scouting<Object>.
+
+ Implementation:
+     * Current runs on top of MINIAOD dataformat of the ScoutingEGammaCollectionMonitoring dataset.
+     * Implemented only for electrons as of now.
+
+Authors: Ting-Hsiang Hsu, Abanti Ranadhir Sahasransu
+*/
+
+#include "ScoutingEGammaCollectionMonitoring.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "DataFormats/Math/interface/LorentzVector.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+
+#include <numeric>
+#include <algorithm>
+
+ScoutingEGammaCollectionMonitoring::ScoutingEGammaCollectionMonitoring(const edm::ParameterSet& iConfig)
+    : outputInternalPath_(iConfig.getParameter<std::string>("OutputInternalPath")),
+      triggerResultsToken_(consumes<edm::TriggerResults>(iConfig.getParameter<edm::InputTag>("TriggerResultTag"))),
+      electronCollection_(
+          consumes<edm::View<pat::Electron>>(iConfig.getParameter<edm::InputTag>("ElectronCollection"))),
+      scoutingElectronCollection_(consumes<std::vector<Run3ScoutingElectron>>(
+          iConfig.getParameter<edm::InputTag>("ScoutingElectronCollection"))),
+      eleIdMapTightToken_(consumes<edm::ValueMap<bool>>(iConfig.getParameter<edm::InputTag>("eleIdMapTight"))) {}
+
+ScoutingEGammaCollectionMonitoring::~ScoutingEGammaCollectionMonitoring() {}
+
+// Function to convert pseudo-rapidity to theta
+double getPtFromEnergyMassEta(double energy, double mass, double eta) {
```

Comment:
same as above. actually if this is needed in multiple classes could be encapsulated in a static utility function elsewhere.

Reply:
@tihsu99: for future code developments, consider checking for usage of static utility function. (See one more comment below.)